aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -447,6 +447,11 @@ Tensor& addmm_out_sparse_csr_cpu(
       mat1.size(1) == mat2.size(0), "mat1 and mat2 shapes cannot be multiplied (",
       mat1.size(0), "x", mat1.size(1), " and ", mat2.sizes()[0], "x", mat2.sizes()[1], ")");
 
+  if (mat1.layout() == kSparseCsc || mat2.layout() == kSparseCsc) {
+    return addmm_out_sparse_compressed_cpu(
+        self, mat1.to_sparse_csr(), mat2.to_sparse_csr(), beta, alpha, result);
```

Comment:
This can be optimized:
```suggestion
    return addmm_out_sparse_compressed_cpu(
        self, mat2.as_sparse_csr(), mat1.as_sparse_csr(), beta, alpha, result).transpose(0, 1);
```
where `as_sparse_csr` would be a view operation that returns a tensor with `SparseCsr` layout but with the same indices and values as the CSC input tensor has.

Consider this as a feature request to introduce `as_sparse_csr/csc/bsr/bsc` methods that linalg ops can take an advantage of.

Reply:
Can't you use transpose for this? It's also a view operation and it gives you the same thing, unless perhaps you want to view say BSC as CSC etc.

Reply:
Indeed, you are right, `as_sparse_*` is equivalent to transpose(..).